// playwright-mocha/dist/cli.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleCLI = void 0;
const commander_1 = __importDefault(require("commander"));
const inquirer_1 = __importDefault(require("inquirer"));
const config_1 = require("./config");
const newConfig = {
    browser: 'chromium',
    testFilesBaseDir: 'spec',
    playwrightLaunchOptions: {},
    playwrightContextOptions: {},
    mochaOptions: {}
};
const generateConfigurationPrompt = [{
        type: 'confirm',
        name: 'configure',
        message: 'Do you want to configure playwright-mocha via CLI?',
        default: true,
    }];
const configQuestions = [
    {
        type: 'list',
        name: 'browser',
        message: 'Please pick you browser',
        choices: ['Chromium', 'Webkit', 'Firefox'],
        filter: (value) => {
            return value.toLowerCase();
        },
    },
    {
        type: 'input',
        name: 'dir',
        message: 'Location of your test specifications?',
    }
];
function handleCLI() {
    return __awaiter(this, void 0, void 0, function* () {
        commander_1.default.parse(process.argv);
        const configPath = './playwright-mocha.json';
        const config = config_1.loadConfig(configPath);
        if (config) {
            return config;
        }
        else {
            const answer = yield inquirer_1.default.prompt(generateConfigurationPrompt);
            if (answer.configure) {
                const configAnswers = yield inquirer_1.default.prompt(configQuestions);
                console.log(configAnswers);
                newConfig.browser = configAnswers.browser;
                newConfig.testFilesBaseDir = configAnswers.dir;
                newConfig.mochaOptions.timeout = 30000;
                config_1.writeConfig(newConfig);
                console.log('Configuration file saved!');
                console.log('To execute your test run npx playwright-mocha!');
                process.exit(0);
            }
            process.exit(0);
        }
    });
}
exports.handleCLI = handleCLI;
